### PR TITLE
Implementing new Exporter & Importer superclasses in ICS stuff 

### PR DESCRIPTION
**Pull request by kenneth-miura**
# Description

Updated names for the ICS exporters + implemented superclasses.

## Type of change

- [x] Bug fix 

# How Has This Been Tested?

manually ran + used pre-existing tests.

# Does this PR follow the rubric?

Using Template. Added to design doc

# Checklist:

- [x] I ran `./gradlew spotlessApply`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I ran `./gradlew build` and the gradle build succeeded
- [x] I have not violated any design principles

# Other Comments:

Other comments that should be made clear.

### CODE REVIEW

**Review by lorena-b — APPROVED** [review]

**Comment by kenneth-miura** [comment]
I'm gonna include the name overload

File: phase2/design_document.md [hunk 1]
```diff
@@ -143,7 +143,9 @@ A Facade is a design pattern that this program uses to simplify the complexity o
 The Template Method is a design pattern that could be used to improve the import/export feature of the program. Currently the program is equipped to serialize schedule data in the .ics file format. If support for other formats was a desired feature, the Template Method could be utilized to define a skeleton of an algorithm that would allow file serialization/deserialization in an abstract sense. Then concrete subclasses (such as `ICSExport` or `CSVExport`) can be designed that would override some parts of the algorithm while retaining the main structure of the algorithm. This would also be an application of the Open-closed principle.
 
 added in [this pr](https://github.com/CSC207-UofT/course-project-tut0201-007/pull/57)
-and used [here](https://github.com/CSC207-UofT/course-project-tut0201-007/pull/62)
+and used [here](https://github.com/CSC207-UofT/course-project-tut0201-007/pull/62).
+
+Further used in [this pr](https://github.com/CSC207-UofT/course-project-tut0201-007/pull/69)
```
Comment (roryhofland) [inline phase2/design_document.md:148]: nice

File: src/main/java/controllers/CommandLineInterface.java [hunk 2]
```diff
@@ -322,7 +325,7 @@ public Schedule displayUserSchedules(List<Schedule> userSchedules) {
                     break;
                 case 'S':
                     System.out.println("Saving this schedule in .ics format...");
-                    ScheduleExporter.outputScheduleICS(currSchedule);
+                    new ICSExporter().outputSchedule(currSchedule);
```
Comment (roryhofland) [inline src/main/java/controllers/CommandLineInterface.java:328]: im guessing the file name option will be added in the refactoring

File: src/main/java/workers/ICSExporter.java [hunk 3]
```diff
@@ -27,30 +23,49 @@
 import util.InvalidSessionException;
 
 /** Use Case class for exporting a Schedule into an ICS file */
-public class ScheduleExporter {
+public class ICSExporter extends Exporter {
 
-    private static int numFiles = 0;
+    private int numFiles = 0;
```
Comment (roryhofland) [inline src/main/java/workers/ICSExporter.java:28]: this is already in the exporter class but its called num, either name works so you can change one or the other

File: src/main/java/workers/ICSExporter.java [hunk 4]
```diff
@@ -27,30 +23,49 @@
...
+    private Calendar calendar;
+    private String baseDir;
+    private File outputDirectory;
 
-    public ScheduleExporter() {}
+    public ICSExporter() {
```
Comment (roryhofland) [inline src/main/java/workers/ICSExporter.java:33]: this all looks good

**Review by roryhofland — CHANGES_REQUESTED** [review]
Just have one small change, rest is bueno

File: src/main/java/controllers/CommandLineInterface.java (hunk 2 above)
Comment (kenneth-miura) [reply src/main/java/controllers/CommandLineInterface.java:328]: yeah, i operated under that assumption. I'm gonna add that as a note to the Ghub projects card.

**Pull request merged**